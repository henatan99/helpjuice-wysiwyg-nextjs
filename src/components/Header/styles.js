import styled from "styled-components";

export const StyledHeaderContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 1.3rem 1.5rem;
`

export const StyledNavWrapper = styled.div`
    display: flex;
`