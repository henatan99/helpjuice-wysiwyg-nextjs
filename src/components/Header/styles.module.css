.interactive:hover {
    cursor: pointer;
}

.interactive:hover * {
    color: '#999999'; 
}