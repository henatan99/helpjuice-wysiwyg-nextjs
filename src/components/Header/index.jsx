import Image from "next/image"
import { StyledHeaderContainer, StyledNavWrapper } from "./styles"
import styles from './styles.module.css'

const HeaderNav = () => {

    const left = [
        {id: 1, content: <Image src='/assets/right-arrow-double.svg' width='10px' height='9px' alt='right-arrow' />, margin: '1.8rem', opacity: 0.4},
        {id: 2, content: <Image src='/assets/book.png' width='15px' height='12.2px' alt='book'/>, margin: '0.6rem'},
        {id: 3, content: "Main", margin: '0.7rem', underlined: true},
        {id: 4, content: "/", margin: '0.7rem', fontSize: '14px'},
        {id: 5, content: "Getting Started", margin: '0.7rem'},
        {id: 6, content: "/", margin: '0.7rem', fontSize: '14px'},
        {id: 7, content: "Front-end developement test proje...", margin: '1rem'},
    ]
    const right = [
        {id: 1, content: <Image src='/assets/lock.png' width='15px' height='16.9px' alt='lock' />, margin: '0.8rem'},
        {id: 2, content: "Editing", margin: '1rem'},
        {id: 3, content: "|", margin: '1rem' ,opacity: 0.6},
        {id: 4, content: "Publish Space", margin: '1rem', color: '#3565a9', fontWeight: 700},
        {id: 5, content: <Image src='/assets/down_expand.svg' width='10px' height='10px' alt='expand' />},
    ]

    return (
        <div
            style={{
                display: 'flex',
                flexSirection: 'row',
                justifyContent: 'space-between',
                alignItems: 'center',
                padding: '1.3rem 1.5rem'
            }}
         >
            <div style={{display: 'flex'}}>
                {
                    left && left.map(elem => {
                        return (
                            <span key={elem.id} style={{
                                marginRight: elem.margin, 
                                display: 'flex', 
                                fontSize: elem.fontSize || '14px', 
                                color: elem.color || '#969ba4', 
                                opacity: elem.opacity,
                                "&:hover": {
                                    cursor: 'pointer',
                                },
                                textDecoration: elem.underlined && 'underline'
                            }}
                            className={elem.id != 4 && elem.id != 6 && styles.interactive}
                            >{elem.content}</span>
                        )
                    })
                }
            </div>
            <div style={{display: 'flex'}}>
                {
                    right && right.map(elem => {
                        return (
                            <span key={elem.id} 
                                className={elem.id != 4  && styles.interactive}
                                style={{
                                    marginRight: elem.margin, 
                                    display: 'flex', 
                                    fontSize: '14px', 
                                    color: elem.color || '#969ba4', 
                                    fontWieght: elem.fontWeight && elem.fontWeight, 
                                    opacity: elem.opacity
                                }}
                            >{elem.content}</span>
                        )
                    })
                }
            </div>
        </div>
    )
}

export default HeaderNav