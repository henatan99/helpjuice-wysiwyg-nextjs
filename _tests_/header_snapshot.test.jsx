import { render } from '@testing-library/react'
import Header from '../src/components/Header'

it('renders header unchanged', () => {
  const { container } = render(<Header />)
  expect(container).toMatchSnapshot()
})