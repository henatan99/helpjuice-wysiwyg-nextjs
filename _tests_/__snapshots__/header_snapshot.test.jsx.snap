// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header unchanged 1`] = `
<div>
  <div
    style="display: flex; justify-content: space-between; align-items: center; padding: 1.3rem 1.5rem;"
  >
    <div
      class="sc-gsnTZi iopORo"
      style="display: flex;"
    >
      <span
        style="margin-right: 1.8rem; display: flex; font-size: 14px; color: rgb(150, 155, 164); opacity: 0.4;"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2710%27%20height=%279%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="right-arrow"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.6rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2712%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="book"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 0.7rem; display: flex; font-size: 14px; color: rgb(150, 155, 164); text-decoration: underline;"
      >
        Main
      </span>
      <span
        style="margin-right: 0.7rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        /
      </span>
      <span
        style="margin-right: 0.7rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        Getting Started
      </span>
      <span
        style="margin-right: 0.7rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        /
      </span>
      <span
        style="margin-right: 1rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        Front-end developement test proje...
      </span>
    </div>
    <div
      class="sc-gsnTZi iopORo"
      style="display: flex;"
    >
      <span
        style="margin-right: 0.8rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2716%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="lock"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
      <span
        style="margin-right: 1rem; display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        Editing
      </span>
      <span
        style="margin-right: 1rem; display: flex; font-size: 14px; color: rgb(150, 155, 164); opacity: 0.6;"
      >
        |
      </span>
      <span
        style="margin-right: 1rem; display: flex; font-size: 14px; color: rgb(53, 101, 169);"
      >
        Publish Space
      </span>
      <span
        style="display: flex; font-size: 14px; color: rgb(150, 155, 164);"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2710%27%20height=%2710%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="expand"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
    </div>
  </div>
</div>
`;
